FEAT(client,avatars): Add avatar icons to user view

GeneralUser01 opened the pull request:
The UI was here changed to use the avatar (if the user has one) as user icon in the user view instead of only showing it in a tooltip when a user is hovered and in the overlay.

This enables one to easier distinguish users as well as getting more use of each user's avatar in a way that would be expected. Below is an example showing the differences.

Without avatar:
![image](https://github.com/user-attachments/assets/fb04e0fa-afa0-45ea-9f4e-7edda48259de)
![image](https://github.com/user-attachments/assets/00e84c20-1cc4-4e72-9961-a1480cea73e0)

With avatar:
![image](https://github.com/user-attachments/assets/ac32a286-766c-497b-bc9e-25bd3b28b115)
![image](https://github.com/user-attachments/assets/48dd2d90-0ac7-4258-ad1a-23925261f6eb)


The "What's this?" message on users was also extended to include the custom avatar talk indicator as shown here:
![image](https://github.com/user-attachments/assets/91cdecef-3dd5-4626-80a1-8a40a2808fd6)


### Checks

- [x] My commits follow the [commit guidelines](https://github.com/mumble-voip/mumble/blob/master/COMMIT_GUIDELINES.md)

GeneralUser01 committed:
FEAT(client,avatars): Add avatar icons to user view

The UI was here changed to use the avatar as user icon in the user view instead of only showing it in a tooltip when a user is hovered, in a comment on them and in the overlay.

This enables one to easier distinguish users as well as getting more use of each user's avatar in a way that would be expected.

Krzmbrzl reviewed (changes requested):
The hardcoded alignmntsaddings also seem like they could cause trouble for users with different screen resolutions.

GeneralUser01 commented on src/mumble/UserView.cpp:
```diff
@@ -21,6 +21,52 @@
 UserDelegate::UserDelegate(QObject *p) : QStyledItemDelegate(p) {
 }
 
+bool UserDelegate::drawAvatarIcon(QPainter *painter, const QRect &rect, ClientUser *user) {
```
Are references always to be preferred over pointers in parameters unless they are optional or to be checked later? I know the codebase has a bunch of approaches which you don't want to be used in new code, so the current code of this application isn't all around a good example. Qt often provides parameters as pointers in its interfaces, but I guess this is for old compatibility/legacy purposes.

Krzmbrzl replied:
Yes - if `nullptr` is not a valid input for that parameter, it should be a reference and not a pointer. This makes it harder to use the function in a non-sensical way as it clearly states that it expects an existing object as a parameter.

I know that Qt uses pointers all over the place but imo that's bad practice 'cause in many cases passing `nullptr` will crash the program because it is not a valid input (and a pointer that must not be null is a reference).

GeneralUser01 replied:
Okay, from here on I would only use a pointer instead of a reference parameter if `nullptr` is a valid input. I would also consider doing so if it is required to pass it as a pointer to another object, but probably would not do so since the argument can simply be converted using the address-of operator (`&`), if it is not done implicitly, while the parameter it comes from more or less ensures it is not null at least.

Krzmbrzl replied:
Exactly.

For passing pointers further along, the argument holds as you wrote it. Getting a pointer from a reference is trivial so making it semantically clear where `nullptr` is allowed and where not is more important imo.

GeneralUser01 commented on src/mumble/UserView.cpp:
```diff
@@ -21,6 +21,52 @@
+	QPen pen;
+	bool isPassive = false;
+	switch (user->tsState) {
+		case Settings::TalkState::Talking:
+			pen.setColor(QColor(68, 163, 242));
+			break;
+		case Settings::TalkState::Whispering:
+			pen.setColor(QColor(142, 83, 167));
+			break;
+		case Settings::TalkState::Shouting:
+			pen.setColor(QColor(229, 172, 7));
+			break;
+		case Settings::TalkState::MutedTalking:
+			pen.setColor(QColor(113, 113, 113));
```
Does the current user icon have colors based on the theme?
If not, why should it differ for avatars?
If so, where are these colors in the theme?

GeneralUser01 replied:
The user icon is the same regardless of theme, at least it appears so by the code I've seen and the user icon I've viewed in lite-, dark- and no theme. The colors for the default avatar are defined in the SVGs they're used, more specifically in the `fill` attribute on each `path` element, with there generally being one SVG per talk state. These are located just under `themes/Default`.

These colors should be defined in a way where both the default avatar SVGs and custom avatars have access to the same definitions, thereby making it easier to track and change the talk status indicator colors consistently. I'm not sure if values defined elsewhere can be applied to SVGs at compile-time but it could be at run-time, not sure where that should be done in that case though. How do you think this should be handled?

Hartmnt replied:
I recently learned that you can set custom properties to Qt elements. At least through the qtcreator, not sure if programmatically. Setting a ``class`` property allows you to specify class-selectors in the css files, which should be exactly what we are looking for here

Krzmbrzl replied:
Yeah, that's also what I did for the TalkingUI, iirc

GeneralUser01 committed:
FIX(client,avatars): Also update user view when removing avatar

Somehow the user view was only updated when setting an avatar and not when removing it. Since `update()` on the widget doesn't work here with `UserDelegate` this change instead refreshes it by adjusting the foreground color back and fourth once. This is done unnoticeably quickly and only adjusts the color by the smallest amount. Furthermore, this would only require a couple repaints unlike a resize which would also require reflows (recalculating the layout).

GeneralUser01 committed:
CHANGE(client,avatars): Make SVGs for custom avatar talk indicator

Because at least the colors of the custom avatar talk indicator should come from the theme, this change makes SVG icons for it in the theme, with there being one for each talk status just as for the default avatar talk indicator except for when passive. The new icons have a transparent background and are overlaid the user-provided image when the user's talk status is not passive.

GeneralUser01 committed:
FIX(client,avatars): Only show icon for listener even when the user has a custom avatar

GeneralUser01 committed:
FEAT(client,avatars): Show custom avatar talk indicator too in "What's this?" message on users

GeneralUser01 committed:
CHANGE(client,avatars): Use predefined color for initializing image with transparency
